### ISSUE TITLE
undefined method `add_measurement' for nil:NilClass

### ISSUE THREAD

**Issue by apotema** [Mjolnir-Software/dynoscale_agent#3]
Hey! 
We've been having some errors with dynoscale_ruby gem on our app...
Version: dynoscale_ruby (1.0.1)
This error was raised 253 times in the 7 days....

Error Stack: 
```
/app/vendor/bundle/ruby/2.7.0/gems/dynoscale_ruby-1.0.1/lib/dynoscale_ruby/recorder.rb:31:in `block (2 levels) in record!'
/app/vendor/bundle/ruby/2.7.0/gems/dynoscale_ruby-1.0.1/lib/dynoscale_ruby/recorder.rb:30:in `each'
/app/vendor/bundle/ruby/2.7.0/gems/dynoscale_ruby-1.0.1/lib/dynoscale_ruby/recorder.rb:30:in `block in record!'
/app/vendor/bundle/ruby/2.7.0/gems/dynoscale_ruby-1.0.1/lib/dynoscale_ruby/recorder.rb:27:in `each'
/app/vendor/bundle/ruby/2.7.0/gems/dynoscale_ruby-1.0.1/lib/dynoscale_ruby/recorder.rb:27:in `record!'
/app/vendor/bundle/ruby/2.7.0/gems/dynoscale_ruby-1.0.1/lib/dynoscale_ruby/middleware.rb:34:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-8.15.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:99:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/activesupport-6.1.7.2/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-8.15.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:99:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.7.2/lib/action_dispatch/middleware/executor.rb:14:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-8.15.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:99:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.7.2/lib/action_dispatch/middleware/static.rb:24:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-8.15.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:99:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.6.3/lib/rack/sendfile.rb:110:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-8.15.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:99:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.7.2/lib/action_dispatch/middleware/host_authorization.rb:142:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-8.15.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:99:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-cors-1.1.1/lib/rack/cors.rb:100:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-8.15.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:99:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/railties-6.1.7.2/lib/rails/engine.rb:539:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-8.15.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:99:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/puma-5.0.4/lib/puma/configuration.rb:239:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/puma-5.0.4/lib/puma/request.rb:76:in `block in handle_request'
/app/vendor/bundle/ruby/2.7.0/gems/puma-5.0.4/lib/puma/thread_pool.rb:337:in `with_force_shutdown'
/app/vendor/bundle/ruby/2.7.0/gems/puma-5.0.4/lib/puma/request.rb:75:in `handle_request'
/app/vendor/bundle/ruby/2.7.0/gems/puma-5.0.4/lib/puma/server.rb:430:in `process_client'
/app/vendor/bundle/ruby/2.7.0/gems/puma-5.0.4/lib/puma/thread_pool.rb:145:in `block in spawn_thread'
```

**Comment by eabraham** [comment]
Hi. Thanks for reporting and including the stack trace. What is the error?

**Comment by apotema** [comment]
Hey, @eabraham 
The error is _NoMethodError_ 
The error message is _undefined method 'add_measurement' for nil:NilClass_

It seems it is being raised on line 31 of the _recorder.rb_ , inside the `self.record!(request_calculator, workers)` method
`@@current_report.add_measurement(current_time, latency, "#{worker.name}:#{queue}", nil)`

**Comment by kpfecteau** [comment]
@eabraham hey it looks like you have pushed a fix for this issue, any chance you will be publishing that soon?

**Comment by eabraham** [comment]
Fixed in v1.0.4
